scaleway/loadbalancers.go

```diff
@@ -1151,6 +1170,73 @@ func (l *loadbalancers) makeCreateBackendRequest(loadbalancer *scwlb.LB, nodePor
 	return request, nil
 }
 
+func (l *loadbalancers) managePrivateNetwork(loadbalancer *scwlb.LB, service *v1.Service) error {
+
+	desiredPrivateNetworks := parsePrivateNetworkAnnotations(service)
+	attachedPrivateNetworks, err := l.api.ListLBPrivateNetworks(
+		&scwlb.ZonedAPIListLBPrivateNetworksRequest{
+			Zone: loadbalancer.Zone,
+			LBID: loadbalancer.ID,
+		}, scw.WithAllPages())
+
+	if err != nil {
+		return err
+	}
+
+	// Detach management
+	detachPrivateNetwork := false
+	for _, attachedPrivatesNetwork := range attachedPrivateNetworks.PrivateNetwork {
+		if !privateNetworkAnnotationsIsPresentAndUpToDate(desiredPrivateNetworks, *attachedPrivatesNetwork) {
+			err = l.api.DetachPrivateNetwork(&scwlb.ZonedAPIDetachPrivateNetworkRequest{
+				Zone:             loadbalancer.Zone,
+				LBID:             loadbalancer.ID,
+				PrivateNetworkID: attachedPrivatesNetwork.PrivateNetworkID,
+			})
+
+			if err != nil {
+				return err
+			}
+			detachPrivateNetwork = true
+		}
+	}
+
+	// Refresh after detach
+	if detachPrivateNetwork {
+		// wait detach is done
+		// TODO: find better solution
```

Comment:
IMHO it should be better to:
- remove from the attachedPrivateNetworks array the private networks that needs to be dettached
- loop on desiredPrivateNetwork 

What do you think?

Reply:
I made changes in https://github.com/scaleway/scaleway-cloud-controller-manager/pull/96/commits/01a38eccd3212f450e31eab13e70dab60242a0f6
What do you think about it ?